### PR TITLE
Added more validations for polling source and http source + documentation

### PR DESCRIPTION
**Pull request by shinitiandrei**
Updated documentation:
- Added a tutorial on how to setup your environment to develop this provider locally. (Feel free to change or add)

Functions added:
- Resource http source: 
  - Added TimeZone validation against Sumologic server
- Resource polling source:
  - All fields in `func resourceSumologicPollingSource() *schema.Resource`  are being validated against Sumologic Server.

File permissions changed

### CODE REVIEW

**Comment by erikvanbrakel** [comment]
I'll take a look at this today. It's a big PR ;-)

File: provider/provider.go [hunk 1]
```diff
@@ -1,48 +1,46 @@
...
+			},
+			"access_key": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  os.Getenv("SL_ACCESSKEY"),
+			},
+			"environment": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "au",
```
Comment (erikvanbrakel) [inline provider/provider.go]: This is a breaking change. Any reason to change this?

File: provider/resource_sumologic_polling_source.go [hunk 2]
```diff
@@ -1,159 +1,188 @@
...
+func thirdyPartyRefSourceAttributes(d *schema.ResourceData, pollingResource []sumologic.PollingResource) error {
+
+	var s []map[string]interface{}
+	for _, t := range pollingResource {
+		mapping := map[string]interface{}{
+			"bucket_name":        t.Path.BucketName,
+			"path_expression":    t.Path.PathExpression,
+		}
+
+		log.Printf("[DEBUG] pollingResources - adding bukets and path expressions: %v", mapping)
```
Comment (erikvanbrakel) [inline provider/resource_sumologic_polling_source.go]: Typo. Not a big deal, just being thorough ;-)

File: provider/provider.go (hunk 1 above)
Comment (erikvanbrakel) [inline provider/provider.go]: Tests fail because of this.

Comment (shinitiandrei) [inline provider/provider.go]: oh sorry, I changed to "au" due to my local tests, putting it back to "eu"

File: provider/resource_sumologic_polling_source.go (hunk 2 above)
Comment (shinitiandrei) [inline provider/resource_sumologic_polling_source.go]: removing it 😅

**Pull request closed without merging**